pkg/cmd/cli/cmd/project.go

```diff
@@ -105,6 +105,15 @@ func (o *ProjectOptions) Complete(f *clientcmd.Factory, args []string, out io.Wr
 
 	o.ClientConfig, err = f.ClientConfig()
 	if err != nil {
+		// if the argument for o.ProjectName passed by the user is a context name,
+		// prevent local context-switching from failing due to an unreachable
+		// server or an unfetchable ClientConfig.
+		if _, exists := o.GetContextFromName(o.ProjectName); exists {
+			o.Config.CurrentContext = o.ProjectName
+			if err := kclientcmd.ModifyConfig(o.PathOptions, o.Config, true); err != nil {
```

Comment:
I think it makes sense, even because you are switching by using the context name here.